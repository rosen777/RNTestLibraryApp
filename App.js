import {SafeAreaView, View, Text, StyleSheet, TextInput} from 'react-native';
import React, {useState} from 'react';

const App = () => {
  const [username, setUsername] = useState('');
  const [userage, setUserage] = useState('');
  const onChangeUserName = name => {
    setUsername(name);
  };
  const onChangeUserAge = age => {
    setUserage(age);
  };

  return (
    <SafeAreaView>
      <View style={styles.wrapper}>
        <Text>Your new username is {username}</Text>
        <TextInput
          style={styles.input}
          onChangeText={text => onChangeUserName(text)}
          value={username}
          placeholder="Username"
          testID="App.username"
        />
        <Text>Your new user age is {userage}</Text>
        <TextInput
          style={styles.input}
          onChangeText={text => onChangeUserAge(text)}
          value={userage}
          placeholder="Userage"
          testID="App.userage"
        />
      </View>
    </SafeAreaView>
  );
};

export default App;

const styles = StyleSheet.create({
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  inputContainer: {
    justifyContent: 'flex-start',
    flexGrow: 1,
  },
});
