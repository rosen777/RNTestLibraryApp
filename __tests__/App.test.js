import React from 'react';
import 'react-native';
import App from '../App';
import {render, fireEvent} from '@testing-library/react-native';

test('renders correctly', () => {
  const {toJSON} = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});

const component = <App />;

describe('Check input field data', () => {
  test('Displays a username, if the username field has been completed', () => {
    const INPUT_TEXT = 'John';
    const {getByTestId, getByText} = render(component);
    const userNameTextInput = getByTestId('App.username');
    fireEvent.changeText(userNameTextInput, INPUT_TEXT);
    const username = getByText(INPUT_TEXT);
    expect(username).toBeTruthy();
  });

  test('Displays a userage, if the userage field has been completed', () => {
    const INPUT_TEXT = '25';
    const {getByTestId, getByText} = render(component);
    const userAgeTextInput = getByTestId('App.userage');
    fireEvent.changeText(userAgeTextInput, INPUT_TEXT);
    const userage = getByText(INPUT_TEXT);
    expect(userage).toBeTruthy();
  });
});
