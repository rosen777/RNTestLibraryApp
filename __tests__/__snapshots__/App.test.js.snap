// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <Text>
      Your new username is 
    </Text>
    <Text>
      
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
      testID="App.username"
      value=""
    />
    <Text>
      Your new user age is 
    </Text>
    <Text>
      
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Userage"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
      testID="App.userage"
      value=""
    />
  </View>
</RCTSafeAreaView>
`;
